retromock/src/main/java/co/infinum/retromock/Retromock.java

```diff
@@ -223,8 +215,59 @@ private Retromock(final Retrofit retrofit,
     }
     final T delegate = factory.create();
 
-    return (T) Proxy.newProxyInstance(service.getClassLoader(), new Class[] {service},
+    return (T) Proxy.newProxyInstance(service.getClassLoader(), new Class[]{service},
       new InvocationHandler() {
+
+        private boolean isKotlinSuspendFunction;
+        private boolean continuationWantsResponse;
+
+        private Object handleContinuationWithResponse(Object[] args, Object call) {
```

Comment:
Let's move code that handles anything regarding coroutines to one class, it will be more readable and we won't need to worry about those boolean variables